include/cryptography/nostr_bech32.h

```diff
@@ -0,0 +1,225 @@
+
+//
+//  nostr_bech32.h
+//  damus
+//
+//  Created by William Casarin on 2023-04-09.
+//
+
+// adapted for aedile-ndk by Finrod Felagund (finrod.felagund.97@gmail.com)
+// a.k.a. npub1ecdlntvjzexlyfale2egzvvncc8tgqsaxkl5hw7xlgjv2cxs705s9qs735
+
+#pragma once
+
+#ifndef nostr_bech32_h
+#define nostr_bech32_h
+
+#include <stdio.h>
+#include <stdbool.h>
+#include <stdint.h>
+
+
+#define likely(x)    __builtin_expect(!!(x), 1)
+#define unlikely(x)  __builtin_expect(!!(x), 0)
+
+#define MAX_RELAYS 10
+
+#define MAX_TLVS 16
+
+#define TLV_SPECIAL 0
+#define TLV_RELAY 1
+#define TLV_AUTHOR 2
+#define TLV_KIND 3
+#define TLV_KNOWN_TLVS 4
+
+#define KEY_LENGTH 32
+#define FROM_BITS 8
+#define TO_BITS 5
+#define CHECKSUM_LENGTH 6
+
+#define MAX_ENCODING_LENGTH 256
+
+
+typedef struct str_block {
+    const char *start;
+    const char *end;
+} str_block_t;
```

Comment:
I won't die on this hill, but stylistically, I'd prefer we denote pointers with the `*` immediately following the type, rather than preceding the name.  E.g., `char* start` rather than `char *start`.

My reasoning for this is that `char* start` is sensibly read as "char-pointer named 'start,'" which is distinct from `char *start` read as "char...pointer-named-'start,'" if that makes sense.

Likewise, `uint8_t *event_id` (for example), represents an array, and writing `uint8_t* event_id` is more akin to `int[] event_id` than the alternative.

```suggestion
typedef struct str_block {
    const char* start;
    const char* end;
} str_block_t;
```